sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -218,15 +234,38 @@ function appendQueryParams(url: string, queryParams: Map<string, string | string
   // QUIRK: parsedUrl.searchParams will have their name/value pairs decoded, which
   // can change their meaning to the server, such as in the case of a SAS signature.
   // To avoid accidentally un-encoding a query param, we parse the key/values ourselves
-  const existingParams = simpleParseQueryParams(parsedUrl.search);
-  const combinedParams = new Map<string, string | string[]>(existingParams);
+  const combinedParams = simpleParseQueryParams(parsedUrl.search);
 
   for (const [name, value] of queryParams) {
     const existingValue = combinedParams.get(name);
     if (Array.isArray(existingValue)) {
-      existingValue.push(...value);
+      if (Array.isArray(value)) {
+        if (existingValue.toString() !== value.toString()) {
+          existingValue.push(...value);
+        }
+      } else {
+        existingValue.push(value);
+      }
     } else if (existingValue) {
-      combinedParams.set(name, [existingValue, ...value]);
+      if (operationSpec.queryParameters?.length) {
+        for (const queryParameter of operationSpec.queryParameters) {
+          if (queryParameter.mapper.serializedName === name) {
+            if (queryParameter.mapper.type.name === "Sequence") {
+              if (Array.isArray(value)) {
+                combinedParams.set(name, [existingValue, ...value]);
```

Comment:
since this is the same logic as on line 260-261 (when the new value is an array, we always create an array with the existing value pre-pended) we can take this out of the inner for loop, right? Also we can use `unshift` on `value` to put `existingValue` in the right spot before setting it.

Reply:
Done